compiler/rustc_codegen_ssa/src/back/write.rs

```diff
@@ -517,6 +518,14 @@ fn copy_all_cgu_workproducts_to_incr_comp_cache_dir(
 
     let _timer = sess.timer("copy_all_cgu_workproducts_to_incr_comp_cache_dir");
 
+    if let Some(path) = metadata.path() {
+        if let Some((id, product)) =
+            copy_cgu_workproduct_to_incr_comp_cache_dir(sess, "metadata", &[("rmeta", path)])
+        {
+            work_products.insert(id, product);
+        }
+    }
```

Comment:
This should be done for all codegen backends. Maybe move it to rustc_incremental or rustc_interface? rustc_interface is responsible for ensuring that the .rmeta file is written in the first place.

Reply:
`rustc_interface` has the `.rmeta` file written by calling `encode_and_write_metadata(tcx)`. This specific code path fetches the metadata to put into the `rlib` file.